htdocs/compta/prelevement/class/bonprelevement.class.php

```diff
@@ -1006,6 +1006,38 @@ public function create($banque = 0, $agence = 0, $mode = 'real', $format = 'ALL'
 		}
 
 		$error = 0;
+		// Pre-store some values intop variables to simplify following sql requests
+		if ($sourcetype != 'salary') {
+			$entity = $type != 'bank-transfer' ? getEntity('invoice') : getEntity('supplier_invoice');
+			$socOrUser = 'fk_soc';
+			$societeOrUser = 'societe';
+		} else {
+			$entity = getEntity('salary');
+			$sqlTable = 'salary';
+			$socOrUser = 'fk_user';
+			$societeOrUser = 'user';
+		}
+
+		$sqlTable = $type != 'bank-transfer' ? "facture" : "facture_fourn";
```

Comment:
These variables have been declared to simplify the following sql requests, by doing only one check on the salary sourcetype and bank-transfert type.
Please note that not everything has been optimized this way, because some simplifications would need the same amount of variable assignation